engine/src/main/java/com/datatorrent/stram/plan/logical/SetupPluginManager.java

```diff
@@ -0,0 +1,130 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.datatorrent.stram.plan.logical;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.slf4j.Logger;
+
+import org.apache.hadoop.conf.Configuration;
+
+import com.datatorrent.stram.StramUtils;
+import com.datatorrent.stram.api.DAGSetupPlugin;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+public class SetupPluginManager
+{
+  private static final Logger LOG = getLogger(SetupPluginManager.class);
+
+  private final transient List<DAGSetupPlugin> plugins = new ArrayList<>();
+  private Configuration conf;
+
+  public static final String DAGSETUP_PLUGINS_CONF_KEY = "org.apache.apex.api";
+  private DAGSetupPlugin.DAGSetPluginContext contex;
+
+  private void loadVisitors(Configuration conf)
+  {
+    LOG.info("loadVisitor called ");
+    this.conf = conf;
+    if (!plugins.isEmpty()) {
+      return;
+    }
+
+    String classNamesStr = conf.get(DAGSETUP_PLUGINS_CONF_KEY);
+    if (classNamesStr == null) {
+      return;
+    }
+    String[] classNames = classNamesStr.split(",");
+    for (String className : classNames) {
+      try {
+        Class<? extends DAGSetupPlugin> plugin = StramUtils.classForName(className, DAGSetupPlugin.class);
+        plugins.add(StramUtils.newInstance(plugin));
+        LOG.info("Found setup plugin {}", plugin);
+      } catch (IllegalArgumentException e) {
+        LOG.warn("Could not load plugin {}", className);
+      }
+    }
+  }
+
+  public void setup(DAGSetupPlugin.DAGSetPluginContext context)
+  {
+    this.contex = context;
+    for (DAGSetupPlugin plugin : plugins) {
+      plugin.setup(context);
+    }
+  }
+
+  public void prePopulateDAG()
```

Comment:
All these method implementations look very similar. How about a common dispatch method that takes a event type.